template/docu/docu_05metaprogramming_02.cc

```diff
@@ -83,19 +83,105 @@ TEST(TemplateMetaprogramming, Reduce) {
             (bricks::metaprogramming::reduce<concatenate_s, std::tuple<A, B, C>>::s()));
 }
     
+template <typename T> struct AsIntImpl {};
+template <> struct AsIntImpl<int> { static int DoIt(int x) { return x; } };
+template <> struct AsIntImpl<const char*> { static int DoIt(const char* s) { return atoi(s); } };
+template <typename T> int AsInt(T x) { return AsIntImpl<T>::DoIt(x); }
+
   // Combine.
+  struct NEG {
+    // A simple way to differentiate logic by struct/class type
+    // is to provide a local, unique, symbol as the 1st param.
```

Comment:
Fixed, in https://github.com/KnowSheet/Bricks/pull/160, https://github.com/dkorolev/Bricks/commit/40b020197aa9b8091443e1a933b70120eabba0cd.